File: src/main/java/platform/ecommerce/dto/response/product/ProductResponse.java
package platform.ecommerce.dto.response.product;

import lombok.Builder;
import platform.ecommerce.domain.product.ProductStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * Product summary response DTO.
 */
@Builder
public record ProductResponse(
        Long id,
        String name,
        String description,
        BigDecimal basePrice,
        Long sellerId,
        Long categoryId,
        ProductStatus status,
        int totalStock,
        String mainImageUrl,
        LocalDateTime createdAt
) {
}

File: src/main/kotlin/platform/ecommerce/dto/response/ProductResponse.kt
package platform.ecommerce.dto.response

import platform.ecommerce.enums.ProductStatus

data class ProductResponse(
    val id: Long,
    val sku: String,
    val name: String,
    val description: String,
    val price: Double,
    val imageUrl: String,
    val status: ProductStatus,
    val options: List<ProductOptionResponse>
)
